ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala

```diff
@@ -18,11 +18,14 @@ class FieldValidations private (errorFactories: ErrorFactories) {
 
   def matchLedgerId(
       ledgerId: LedgerId
-  )(received: LedgerId)(implicit
+  )(receivedO: Option[LedgerId])(implicit
       contextualizedErrorLogger: ContextualizedErrorLogger
-  ): Either[StatusRuntimeException, LedgerId] =
-    if (ledgerId == received) Right(received)
-    else Left(ledgerIdMismatch(ledgerId, received, definiteAnswer = Some(false)))
+  ): Either[StatusRuntimeException, Option[LedgerId]] =
+    receivedO.fold[Either[StatusRuntimeException, Option[LedgerId]]](Right(None)) {
+      received: LedgerId =>
+        if (received == ledgerId) Right(Some(received))
+        else Left(ledgerIdMismatch(ledgerId, received, definiteAnswer = Some(false)))
+    }
```

Comment:
```suggestion
  ): Either[StatusRuntimeException, Option[LedgerId]] = receivedO match {
    case None => Right(None)
    case Some(received) if received == ledgerId => Right(Some(received))
    case Some(_) =>  Left(ledgerIdMismatch(ledgerId, received, definiteAnswer = Some(false))
  }
```

Case distinctions can be easier to read for both user and compiler (no type annotations required) ;-)

Reply:
As a further improvement you can use backticks to match for equality to an existing bindings as follows:
```scala
    case Some(`ledgerId`) => Right(Some(received))
    case Some(mismatching) => Left(ledgerIdMismatch(ledgerId, mismatching, definiteAnswer = Some(false))
```
I also included the following line because if I'm not mistaken that does not compile unless you bind it in the case.

Reply:
Ok, used backticks